samples/subsys/power/stm32_pm/lptim/src/main.c

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) 2016 Intel Corporation.
+ * Copyright (c) 2019 STMicroelectronics.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "sample.h"
+
+struct device *gpio_port, *gpio_port0;
+
+/* 1000 msec = 1 sec */
+
+#define WAIT_TIME 100
+#define SLEEP_TIME 2001
+
+/* measurement gives TIME VALUE +- 0.1ms */
+
+static struct gpio_callback gpio_cb;
+static u32_t timeout = WAIT_TIME;
+
+void button_pressed(struct device *gpiob, struct gpio_callback *cb,
+		    u32_t pins)
+{
+		LL_PWR_ClearFlag_WU();
+		timeout = timeout + WAIT_TIME;
+}
+
+/* Application main Thread */
+void main(void)
+{
+
+	printk("\n\n*** LPTIM Management Demo on %s ***\n", CONFIG_BOARD);
+
+	gpio_port = device_get_binding(PORT);
+
+	/* Configure Button 1 to increase timmeout value (no Pull up on PA0) */
+	gpio_pin_configure(gpio_port, SW,
+			   GPIO_DIR_IN | GPIO_INT |
+			   GPIO_INT_EDGE | GPIO_INT_ACTIVE_LOW);
+
+	gpio_init_callback(&gpio_cb, button_pressed, BIT(SW));
+
+	gpio_add_callback(gpio_port, &gpio_cb);
+	gpio_pin_enable_callback(gpio_port, SW);
+
+	/* Configure LEDs */
+	gpio_port0 = device_get_binding(PORT0);
+	gpio_pin_configure(gpio_port0, LED0, GPIO_DIR_OUT);
+	gpio_pin_write(gpio_port0, LED0, LED_OFF);
+
+	/*
+	 * Start the demo.
+	 */
+
+	while (1) {
+		/* Set pin to HIGH/LOW */
+		gpio_pin_write(gpio_port0, LED0, LED_ON);
+
+		k_sleep(timeout);
```

Comment:
Let's add use of `k_uptime` and `k_uptime_delta` to see if we manage to remain coherent

Reply:
done in main.c function button_pressed